libs/mock-block-dock/src/datastore/traverse.ts

```diff
@@ -1,284 +1,539 @@
 import {
   EntityId,
+  EntityValidInterval,
   GraphElementVertexId,
+  GraphResolveDepths,
   HasLeftEntityEdge,
   HasRightEntityEdge,
   IncomingLinkEdge,
+  isEntityVertex,
+  KnowledgeGraphOutwardEdge,
+  OntologyRootedEdges,
   OutgoingLinkEdge,
+  OutwardEdge,
   Subgraph,
+  SubgraphRootType,
+  TimeInterval,
+  Vertex,
 } from "@blockprotocol/graph";
 import { addKnowledgeGraphEdgeToSubgraphByMutation } from "@blockprotocol/graph/internal";
 import {
   getIncomingLinksForEntity,
   getLeftEntityForLinkEntity,
   getOutgoingLinksForEntity,
+  getRightEntityForLinkEntity,
+  intervalIntersectionWithInterval,
+  mapElementsIntoRevisions,
 } from "@blockprotocol/graph/stdlib";
-import { getRightEntityForLinkEntity } from "@blockprotocol/graph/stdlib-temporal";
 
-import { typedEntries } from "../util";
-import { ResolveMap } from "./traverse/resolve-map";
-import { PartialDepths, TraversalContext } from "./traverse/traversal-context";
+import { get, mustBeDefined, typedEntries } from "../util";
 
-/** @todo - doc */
-/** @todo - Update this to take an interval instead of always being "latest" */
-/** @todo - Update this to handle ontology edges */
-/** @todo - Update this to use temporal versioning information */
-export const traverseElement = (
-  traversalSubgraph: Subgraph,
-  elementIdentifier: GraphElementVertexId,
-  datastore: Subgraph,
-  traversalContext: TraversalContext,
-  currentTraversalDepths: PartialDepths,
-) => {
-  const unresolvedDepths = traversalContext.insert(
-    elementIdentifier,
-    currentTraversalDepths,
+const TIMESTAMP_PLACEHOLDER = `TIMESTAMP_PLACEHOLDER` as const;
```

Comment:
I had backticks because I was going to add `Date.new().toISOString()` (or some other unique identifier) within the template, and decided against the need for it. Will remove, thanks for highlighting

Reply:
[Remove unnecessary template string](https://github.com/blockprotocol/blockprotocol/pull/931/commits/c02858e576fef2ce43c3d50b4b16c6995abb23a3)